### PR TITLE
Various updates

### PR DESCRIPTION
**Pull request by JensTrier**
Updating *-CsOnlineApplicationInstance with new syntax, references and mention 3rd party application id's.

Adding documentation for Extended Notifications for Teams emergency calling policy.

Adding more information about GroupNotificationOverride to Set-CsUserCallingSettings

### CODE REVIEW

**Comment by yogkumgit** [comment]
@JohanFreelancer9

File: skype/skype-ps/skype/Get-CsOnlineApplicationInstance.md [hunk 1]
```diff
@@ -81,14 +81,14 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
-### -Tenant
-The Tenant ID.
+### -Skip
+Skips the first X returned results and the default value is 0.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Get-CsOnlineApplicationInstance.md:85]:
```suggestion
Skips the first specified number of returned results. The default value is 0.
```

File: skype/skype-ps/skype/New-CsOnlineApplicationInstance.md [hunk 2]
```diff
@@ -58,13 +52,13 @@ Accept wildcard characters: False
 ```
 
 ### -ApplicationId
-The application ID.
+The application ID. The Microsoft application Auto Attendant has the ApplicationId ce933385-9390-45d1-9512-c8d228074e07 and the Microsoft application Call Queue has the ApplicationId 11cd3e2e-fccb-42ad-ad00-878b93575e07. Third party applications available in a tenant will use other ApplicationId's.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/New-CsOnlineApplicationInstance.md:55]:
```suggestion
The application ID. The Microsoft application Auto Attendant has the ApplicationId ce933385-9390-45d1-9512-c8d228074e07 and the Microsoft application Call Queue has the ApplicationId 11cd3e2e-fccb-42ad-ad00-878b93575e07. Third-party applications available in a tenant will use other ApplicationId's.
```

File: skype/skype-ps/skype/New-CsTeamsEmergencyCallingExtendedNotification.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+## SYNTAX
+
+```
+New-CsTeamsEmergencyCallingExtendedNotification -EmergencyDialString <string> [-NotificationGroup <string>] 
+ [-NotificationDialOutNumber <string>] [-NotificationMode <string>] [<CommonParameters>]
+```
+
+## DESCRIPTION
+This cmdlet supports creating specific emergency calling notification settings per emergency phone number. Used with TeamsEmergencyCallingPolicy.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/New-CsTeamsEmergencyCallingExtendedNotification.md:25]:
```suggestion
This cmdlet supports creating specific emergency calling notification settings per emergency phone number. It is used with TeamsEmergencyCallingPolicy.

```

File: skype/skype-ps/skype/New-CsTeamsEmergencyCallingExtendedNotification.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+## EXAMPLES
+
+### Example 1
+```powershell
+PS C:>  $en1 = New-CsTeamsEmergencyCallingExtendedNotification -EmergencyDialString "911" -NotificationGroup "alert2@contoso.com" -NotificationMode ConferenceUnMuted
+```
+
+Creates an extended emergency calling notfication for emergency phone number 911 and stores it in the variable $en1. The variable should afterwards be added to a TeamsEmergencyCallingPolicy instance.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/New-CsTeamsEmergencyCallingExtendedNotification.md:34]:
```suggestion
Creates an extended emergency calling notification for the emergency phone number 911 and stores it in the variable $en1. The variable should be added afterward to a TeamsEmergencyCallingPolicy instance.

```

File: skype/skype-ps/skype/New-CsTeamsEmergencyCallingExtendedNotification.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -NotificationGroup
+NotificationGroup is a email list of users and groups to be notified of an emergency call. Individual users or groups are separated by ";", for instance "group1@contoso.com;group2@contoso.com". A maximum of 50 users can be notified.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/New-CsTeamsEmergencyCallingExtendedNotification.md:54]:
```suggestion
NotificationGroup is an email list of users and groups to be notified of an emergency call. Individual users or groups are separated by ";", for instance, "group1@contoso.com;group2@contoso.com". A maximum of 50 users can be notified.

```

File: skype/skype-ps/skype/New-CsTeamsEmergencyCallingPolicy.md [hunk 6]
```diff
@@ -29,17 +29,34 @@ This cmdlet creates a new Teams Emergency Calling policy. Emergency calling poli
...
+This example creates a Teams Emergency Calling policy that has an identity of TestECP, where a notification group and number is being defined, the external location lookup mode is enabled and also the type of notification.
 
 ### Example 2
 ```powershell
-PS C:>  New-CsTeamsEmergencyCallingPolicy -Identity "testECP2" -NotificationGroup "123@gh.com;567@test.com"
+New-CsTeamsEmergencyCallingPolicy -Identity "TestECP2" -NotificationGroup "123@contoso.com;567@contoso.com"
 ```
 
- This example creates a Teams Emergency Calling policy that has an identity of testECP2, with default settings except for the Notification Group. This parameter expects a single string with all users and groups separated with ";".
+This example creates a Teams Emergency Calling policy that has an identity of TestECP2, with default settings except for the Notification Group. This parameter expects a single string with all users and groups separated with ";".
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/New-CsTeamsEmergencyCallingPolicy.md:42]:
```suggestion
This example creates a Teams Emergency Calling policy that has an identity of TestECP2, with default settings, except for the Notification Group. This parameter expects a single string with all users and groups separated by ";".
```

File: skype/skype-ps/skype/New-CsTeamsEmergencyCallingPolicy.md [hunk 7]
```diff
@@ -29,17 +29,34 @@ This cmdlet creates a new Teams Emergency Calling policy. Emergency calling poli
...
+
+### Example 3
+```powershell
+$en1 = New-CsTeamsEmergencyCallingExtendedNotification -EmergencyDialString "112" -NotificationGroup "alert2@contoso.com" -NotificationMode ConferenceUnMuted
+$en2 = New-CsTeamsEmergencyCallingExtendedNotification -EmergencyDialString "911" -NotificationGroup "alert3@contoso.com" -NotificationMode NotificationOnly -NotificationDialOutNumber "+14255551234"
+Set-CsTeamsEmergencyCallingPolicy -Identity "TestECP3" -ExtendedNotifications @{add=$en1,$en2}
+```
+
+This example creates specifc emergency calling notification settings for two emergency phone numbers and adds them to the existing TestECP3 policy instance.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/New-CsTeamsEmergencyCallingPolicy.md:51]:
```suggestion
This example creates specific emergency calling notification settings for two emergency phone numbers and adds them to the existing TestECP3 policy instance.
```

File: skype/skype-ps/skype/New-CsTeamsEmergencyCallingPolicy.md [hunk 8]
```diff
@@ -29,17 +29,34 @@ This cmdlet creates a new Teams Emergency Calling policy. Emergency calling poli
...
+
+### Example 4
+```powershell
+$en2 = New-CsTeamsEmergencyCallingExtendedNotification -EmergencyDialString "911" -NotificationGroup "alert3@contoso.com" -NotificationMode NotificationOnly -NotificationDialOutNumber "+14255551234"
+Set-CsTeamsEmergencyCallingPolicy -Identity "TestECP3" -ExtendedNotifications @{remove=$en2}
+```
+
+This example removes a specifc emergency calling notification setting from the existing TestECP3 policy instance.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/New-CsTeamsEmergencyCallingPolicy.md:59]:
```suggestion
This example removes a specific emergency calling notification setting from the existing TestECP3 policy instance.
```

File: skype/skype-ps/skype/New-CsTeamsEmergencyCallingPolicy.md [hunk 9]
```diff
@@ -105,7 +139,7 @@ Accept wildcard characters: False
 ```
 
 ### -NotificationDialOutNumber
-This parameter represents the PSTN number which can be dialed out if NotificationMode is set to either of the two Conference values.
+This parameter represents the PSTN number which can be dialed out if NotificationMode is set to either of the two Conference values.  The PSTN phone cannot be unmuted even when the NotificationMode is set to ConferenceUnMuted.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/New-CsTeamsEmergencyCallingPolicy.md:142]:
```suggestion
This parameter represents the PSTN number which can be dialed out if NotificationMode is set to either of the two Conference values. The PSTN phone cannot be unmuted even when the NotificationMode is set to ConferenceUnMuted.
```

File: skype/skype-ps/skype/Set-CsOnlineApplicationInstance.md [hunk 10]
```diff
@@ -21,27 +21,20 @@ and [Remove-CsPhoneNumberAssignment](/powershell/module/teams/remove-csphonenumb
...
+Set-CsOnlineApplicationInstance -Identity appinstance01@contoso.com -ApplicationId ce933385-9390-45d1-9512-c8d228074e07 -DisplayName "AppInstance01"
 ```
 
-This example shows updating OnpremPhoneNumber, ApplicationId, DisplayName information for an existing Auto Attendant application instance with Identity "appinstance01@contoso.com".
-
-The following are the application ID's for each type of application instance types:
-
-Auto Attendant: ce933385-9390-45d1-9512-c8d228074e07
-Call Queue: 11cd3e2e-fccb-42ad-ad00-878b93575e07
+This example shows updating ApplicationId and DisplayName information for an existing Auto Attendant application instance with Identity "appinstance01@contoso.com".
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsOnlineApplicationInstance.md:37]:
```suggestion
This example shows updated ApplicationId and DisplayName information for an existing Auto Attendant application instance with Identity "appinstance01@contoso.com".
```

File: skype/skype-ps/skype/Set-CsOnlineApplicationInstance.md [hunk 11]
```diff
@@ -80,13 +73,13 @@ Accept wildcard characters: False
 ```
 
 ### -ApplicationId
-The application ID.
+The application ID. The Microsoft application Auto Attendant has the ApplicationId ce933385-9390-45d1-9512-c8d228074e07 and the Microsoft application Call Queue has the ApplicationId 11cd3e2e-fccb-42ad-ad00-878b93575e07. Third party applications available in a tenant will use other ApplicationId's.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsOnlineApplicationInstance.md:76]:
```suggestion
The application ID. The Microsoft application Auto Attendant has the ApplicationId ce933385-9390-45d1-9512-c8d228074e07 and the Microsoft application Call Queue has the ApplicationId 11cd3e2e-fccb-42ad-ad00-878b93575e07. Third-party applications available in a tenant will use other ApplicationId's.
```

File: skype/skype-ps/skype/Set-CsTeamsEmergencyCallingPolicy.md [hunk 12]
```diff
@@ -30,10 +31,27 @@ This cmdlet modifies an existing Teams Emergency Calling policy. Emergency calli
...
+This example modifies an existing policy instance with identity TestECP.
+
+### Example 2
+```powershell
+$en1 = New-CsTeamsEmergencyCallingExtendedNotification -EmergencyDialString "112" -NotificationGroup "alert2@contoso.com" -NotificationMode ConferenceUnMuted
+$en2 = New-CsTeamsEmergencyCallingExtendedNotification -EmergencyDialString "911" -NotificationGroup "alert3@contoso.com" -NotificationMode NotificationOnly -NotificationDialOutNumber "+14255551234"
+Set-CsTeamsEmergencyCallingPolicy -Identity "TestECP" -ExtendedNotifications @{add=$en1,$en2}
+```
+
+This example creates specifc emergency calling notification settings for two emergency phone numbers and adds them to the existing TestECP policy instance.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsTeamsEmergencyCallingPolicy.md:46]:
```suggestion
This example creates specific emergency calling notification settings for two emergency phone numbers and adds them to the existing TestECP policy instance.
```

File: skype/skype-ps/skype/Set-CsTeamsEmergencyCallingPolicy.md [hunk 13]
```diff
@@ -30,10 +31,27 @@ This cmdlet modifies an existing Teams Emergency Calling policy. Emergency calli
...
+
+### Example 3
+```powershell
+$en2 = New-CsTeamsEmergencyCallingExtendedNotification -EmergencyDialString "911" -NotificationGroup "alert3@contoso.com" -NotificationMode NotificationOnly -NotificationDialOutNumber "+14255551234"
+Set-CsTeamsEmergencyCallingPolicy -Identity "TestECP" -ExtendedNotifications @{remove=$en2}
+```
+
+This example removes a specifc emergency calling notification setting from the existing TestECP policy instance.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsTeamsEmergencyCallingPolicy.md:54]:
```suggestion
This example removes a specific emergency calling notification setting from the existing TestECP policy instance.
```

File: skype/skype-ps/skype/Set-CsTeamsEmergencyCallingPolicy.md [hunk 14]
```diff
@@ -67,11 +85,28 @@ Accept pipeline input: False
...
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
 ### -ExternalLocationLookupMode
-Enables ExternalLocationLookupMode. This mode allows users to set Emergency addresses for remote locations.
+Enable ExternalLocationLookupMode. This mode allow users to set Emergency addresses for remote locations.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Set-CsTeamsEmergencyCallingPolicy.md:106]:
```suggestion
Enables ExternalLocationLookupMode. This mode allows users to set Emergency addresses for remote locations.
```

File: skype/skype-ps/skype/Sync-CsOnlineApplicationInstance.md [hunk 15]
```diff
@@ -76,7 +75,7 @@ Accept wildcard characters: False
 ```
 
 ### -ApplicationId
-The application ID.
+The application ID. The Microsoft application Auto Attendant has the ApplicationId ce933385-9390-45d1-9512-c8d228074e07 and the Microsoft application Call Queue has the ApplicationId 11cd3e2e-fccb-42ad-ad00-878b93575e07. Third party applications available in a tenant will use other ApplicationId's.
```
Comment (JohanFreelancer9) [inline skype/skype-ps/skype/Sync-CsOnlineApplicationInstance.md:78]:
```suggestion
The application ID. The Microsoft application Auto Attendant has the ApplicationId ce933385-9390-45d1-9512-c8d228074e07 and the Microsoft application Call Queue has the ApplicationId 11cd3e2e-fccb-42ad-ad00-878b93575e07. Third-party applications available in a tenant will use other ApplicationId's.
```

File: teams/teams-ps/teams/Set-CsUserCallingSettings.md [hunk 16]
```diff
@@ -270,7 +270,7 @@ Accept wildcard characters: False
 
 The group notification override that will be set on the specified user. The supported values are Ring, Mute and Banner.
 
-Setting this overrides the call group notification specified for the individual call group for the specified user.
+The initial setting is shown as Null. It means that the group notification set for the user in the call group is used. If you set GroupNotificationOverride to Mute, that setting will override the group notification for the user in the call group for the user. If you set the GroupNotificationOverride to Ring or Banner, the group notification set for the user in the call group will be used.
```
Comment (JohanFreelancer9) [inline teams/teams-ps/teams/Set-CsUserCallingSettings.md:273]:
```suggestion
The initial setting is shown as Null. It means that the group notification set for the user in the call group is used. If you set GroupNotificationOverride to Mute, that setting will override the group notification for the user in the call group. If you set the GroupNotificationOverride to Ring or Banner, the group notification set for the user in the call group will be used.

```

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
Copy review - @JensTrier, kindly commit the suggested changes. Many thanks.

**Comment by JensTrier** [comment]
> Copy review - @JensTrier, kindly commit the suggested changes. Many thanks.

Thanks! Done

**Review by JohanFreelancer9 — APPROVED** [review]

**Review by dariomws — APPROVED** [review]

**Comment by JensTrier** [comment]
@JohanFreelancer9 @tseward @dariomws Sorry for the extra work, but I updated the PR with 2 new examples in Get-CsPhoneNumberAssignment. Please have a look.

**Comment by yogkumgit** [comment]
@get-itips Please review. Thanks

**Review by get-itips — APPROVED** [review]

**Comment by yogkumgit** [comment]
@tiburd Please merge this PR. Thanks

**Pull request merged**